ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -412,6 +412,41 @@ public Collection<Container> updateContainers(String accountName, Collection<Con
     }
   }
 
+  @Override
+  public Dataset addDatasetVersion(String accountName, String containerName, String datasetName, String version,
+      long expirationTimeMs) throws AccountServiceException {
+    try {
+      Container container = getContainerByName(accountName, containerName);
+      if (container == null) {
+        throw new AccountServiceException("Can't find the container: " + containerName + " in account: " + accountName,
+            AccountServiceErrorCode.BadRequest);
+      }
+      short accountId = container.getParentAccountId();
+      short containerId = container.getId();
+      return mySqlAccountStore.addDatasetVersion(accountId, containerId, accountName, containerName, datasetName,
+          version, expirationTimeMs);
```

Comment:
we have in memory cache for account and container, do we want to cache dataset as well?